eclipse/plugins/org.switchyard.tools.ui/src/org/switchyard/tools/ui/common/SwitchYardSettingsGroup.java

```diff
@@ -591,6 +591,9 @@ private void handleRuntimeSelected() {
             final ArtifactVersion version = getRuntimeVersion((IRuntimeComponent) selection.getFirstElement());
             if (version != null) {
                 _runtimeVersionsList.setSelection(new StructuredSelection(version));
+                
+                // to address SWITCHYARD-2968, force the text for Mac OS/X or doesn't get set on selection
+                _runtimeVersionsList.getCombo().setText(version.toString());
```

Comment:
setSelection twice doesn't do anything in this case. setting the text on the control seems the only way to resolve it. I suspect you are correct on listeners stepping on one another, but as this simple solution works to fix the problem I'm going to stick with it at this point since it only affects Mac.